src/System.Slices/src/System/Span.cs

```diff
@@ -13,14 +12,14 @@ namespace System
     /// to regular accesses and is a struct so that creation and subslicing do
     /// not require additional allocations.  It is type- and memory-safe.
     /// </summary>
-    public struct Span<T> : IEnumerable<T>, IEquatable<Span<T>>
+    public partial struct Span<T> : IEnumerable<T>, IEquatable<Span<T>>
     {
         /// <summary>A managed array/string; or null for native ptrs.</summary>
         readonly object _object;
         /// <summary>An byte-offset into the array/string; or a native ptr.</summary>
         readonly UIntPtr _offset;
         /// <summary>Fetches the number of elements this Span contains.</summary>
-        public int Length { get; private set; }
+        public readonly int Length;
```

Comment:
cc @krzysztofcwalina, it would be great to investigate why. But ok, let's use as a field first.